jsonnet/main.jsonnet

```diff
@@ -320,7 +320,27 @@ local inCluster =
     },
 
     // Objects
-    clusterMonitoringOperator: clusterMonitoringOperator($.values.clusterMonitoringOperator),
+    clusterMonitoringOperator: clusterMonitoringOperator($.values.clusterMonitoringOperator) {
+      // The cluster-monitoring-operator ClusterRole needs the combined set of
+      // permissions from all its operand ClusterRoles.  This extends the base
+      // ClusterRole by just appending the rules from the others.
+      clusterRole+: {
+        rules+: inCluster.alertmanager.clusterRole.rules +
+                inCluster.clusterMonitoringOperator.clusterRoleView.rules +
+                inCluster.grafana.clusterRole.rules +
+                inCluster.kubeStateMetrics.clusterRole.rules +
+                inCluster.nodeExporter.clusterRole.rules +
+                inCluster.openshiftStateMetrics.clusterRole.rules +
+                inCluster.prometheusAdapter.clusterRole.rules +
+                inCluster.prometheusAdapter.clusterRoleAggregatedMetricsReader.rules +
+                inCluster.prometheusAdapter.clusterRoleServerResources.rules +
+                inCluster.prometheus.clusterRole.rules +
+                inCluster.prometheusOperator.clusterRole.rules +
+                inCluster.telemeterClient.clusterRole.rules +
+                inCluster.thanosQuerier.clusterRole.rules +
+                inCluster.thanosRuler.clusterRole.rules,
```

Comment:
It looks like we also need to add `Role` objects and not only `ClusterRole` ones - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/1230/pull-ci-openshift-cluster-monitoring-operator-master-e2e-agnostic-operator/1405518486192525312#1:build-log.txt%3A51

Reply:
Seems like this warning is valid in our case: https://github.com/openshift/cluster-monitoring-operator/blob/master/hack/merge_cluster_roles.py#L57-L59

```console
$ make manifests/0000_50_cluster-monitoring-operator_02-role.yaml
python2 hack/merge_cluster_roles.py hack/cluster-monitoring-operator-role.yaml.in `find assets | grep role | grep -v "role-binding"` > manifests/0000_50_cluster-monitoring-operator_02-role.yaml
hack/merge_cluster_roles.py:61: UserWarning: creating a ClusterRole from a Role
  warnings.warn('creating a ClusterRole from a {}'.format(self.manifest['kind']))
```

Reply:
Ah, good call. I actually meant to do that and just forgot.